lib/symbol_count_graph.rb

```diff
@@ -1,10 +1,11 @@
 class SymbolCountGraph
-  attr_reader :data, :options
+  attr_reader :data, :options, :prepared_data
 
   def initialize(data, options = { columns: 10 })
     validate_arguments(data, options)
     @data = data
     @options = options
+    prepare_data
   end
 
   def render
```

Comment:
es gibt jetzt zwei verschiedene ausgabe formate a)  `String` “xx\nxo\no” und b) 2.dimensionaler `Array` [ ['x', ‘x’], ['x', ‘’],[‘o’] ]
#render generiert den string und prepare_data den array wäre es nicht besser, wenn es eine methode gibt die die struktur generiert und eine andere die diese dann in ein anderes format übersetzt?